docs/Displaying a User Interface/Alerts/index.md

```diff
@@ -1,135 +1,217 @@
-alert.duration = @5000;
-
-// A progress indicator (e.g. spinning wheel or hourglass)
-// Not all head units support the progress indicator
-alert.progressIndicator = @YES;
+NSMutableArray<SDLSoftButton *> *softButtons = [[NSMutableArray alloc] init];
 
-// Text-to-speech
-alert.ttsChunks = [SDLTTSChunk textChunksFromString:@"<#Text to speak#>"];
-
-// Special tone played before the tts is spoken
-alert.playTone = @YES;
+SDLSoftButton *button1 = [[SDLSoftButton alloc] initWithType:SDLSoftButtonTypeText text:<#Button Text#> image:nil highlighted:@NO buttonId:<#Soft Button Id#> systemAction:SDLSystemActionDefaultAction handler:^(SDLOnButtonPress *_Nullable buttonPress, SDLOnButtonEvent *_Nullable buttonEvent) {
+    if (buttonPress == nil) {
+        return;
+    }
+    <#Code for when the button is pressed#>
+}];
 
-// Soft buttons
-SDLSoftButton *okButton = [[SDLSoftButton alloc] init];
-okButton.text = @"OK";
-okButton.type = SDLSoftButtonTypeText;
-okButton.softButtonID = @<#Soft Button Id#>;
-okButton.handler = ^(SDLOnButtonPress *_Nullable buttonPress,  SDLOnButtonEvent *_Nullable buttonEvent) {
+SDLSoftButton *button2 = [[SDLSoftButton alloc] initWithType:SDLSoftButtonTypeText text:<#Button Text#> image:nil highlighted:@NO buttonId:<#Soft Button Id#> systemAction:SDLSystemActionDefaultAction handler:^(SDLOnButtonPress *_Nullable buttonPress, SDLOnButtonEvent *_Nullable buttonEvent) {
     if (buttonPress == nil) {
-      return;
+        return;
     }
+    <#Code for when the button is pressed#>
+}];
 
-    // create a custom action for the selected button
-};
+[softButtons addObject: button1];
+[softButtons addObject: button2];
 
-alert.softButtons = @[okButton];
+alert.softButtons = softButtons;
+```
 
-// Send the alert
-[self.sdlManager sendRequest:alert withResponseHandler:^(SDLRPCRequest *request, SDLRPCResponse *response, NSError *error) {
-    if ([response.resultCode isEqualToEnum:SDLResultSuccess]) {
-      // alert was dismissed successfully
-    }
-}];
+#### Swift
+```swift
+let alert = SDLAlert(alertText1: "<#Line 1#>", alertText2: "<#Line 2#>", alertText3: "<#Line 3#>")
+
+var softButtons = [SDLSoftButton]()
+
+let button1 = SDLSoftButton(type: .text, text: <#Button Text#>, image: nil, highlighted: false, buttonId: <#Soft Button Id#>, systemAction: .defaultAction, handler: { buttonPress, buttonEvent in
+    guard buttonPress != nil else { return }
+    <#Code for when the button is pressed#>
+})
+
+let button2 = SDLSoftButton(type: .text, text: <#Button Text#>, image: nil, highlighted: false, buttonId: <#Soft Button Id#>, systemAction: .defaultAction, handler: { buttonPress, buttonEvent in
+    guard buttonPress != nil else { return }
+    <#Code for when the button is pressed#>
+})
+
+softButtons.append(button1)
+softButtons.append(button2)
+
+alert.softButtons = softButtons;
+```
+!@
+
+@![android,javaSE,javaEE]
+` TODO - code example `
+!@
+
+## Additonal Alert Paramaters
+
```

Comment:
Add a small code example for just creating the alert with text.

Reply:
moved the section above to do this